source/tutorials/kubernetes.rst

```diff
@@ -3,170 +3,340 @@
+
+
+#. Enable the kubelet agent service to start at boot automatically:
 
    .. code-block:: bash
 
       sudo systemctl enable kubelet.service
 
-#. Disable swap using one of the following methods, either:
 
-   a) Temporarily:
+Initialize the master node
+**************************
+
+In Kubernetes, a master node is part of the `Kubernetes Control Plane
+<https://kubernetes.io/docs/concepts/#kubernetes-control-plane>`_. 
 
-      .. code-block:: bash
+Initializing a new Kubernetes cluster involves crafting a :command:`kubeadm
+init` command. Adding parameters to this command can control the fundamental
+operating components of the cluster. This means it is important to understand
+and choose network and runtime options before running a :command:`kubeadm
+init` command.
 
-         sudo swapoff -a
 
-      .. note::
+Choose a pod network add-on
+===========================
 
-         Swap will be enabled at next reboot, causing failures in
-         your cluster if you choose this method.
+In Kubernetes, a `pod
+<https://kubernetes.io/docs/concepts/workloads/pods/pod/>`_ is a group of one
+or more containers and is the smallest deployable unit of computing in a
+Kubernetes cluster. Pods have shared storage/network internally but
+communication between pods requires additional configuration. If you want your
+pods to be able to communicate with each other you must choose and install a
+`pod network add-on`_. Otherwise, this section can be skipped. 
 
-   or:
+This is important to decide early because some pod network add-ons require
+configuration during cluster initialization. Check whether or not your add-on
+requires special flags when you initialize the master control plane.
 
-   b) Permanently:
+If your chosen network add-on requires appending to the :command:`kubeadm
+init` command, make note of it before continuing. 
 
-      Mask the swap partition:
+For example, if you choose the *flannel* pod network add-on, then in later
+steps you must add the following to the :command:`kubeadm init` command:
 
-      .. code-block:: bash
+.. code-block:: console
+
+   --pod-network-cidr 10.244.0.0/16
 
-         sudo systemctl mask $(sed -n -e 's#^/dev/\([0-9a-z]*\).*#dev-\1.swap#p' /proc/swaps) 2>/dev/null
-         sudo swapoff -a
 
-      .. note::
 
-         On systems with limited resources, some performance degradation may
-         be observed while swap is disabled.
+Choose a container runtime
+==========================
 
-#. Switch to root to modify the `hosts` file:
+Container runtimes are the underlying fabric that pod workloads execute inside
+of. Different container runtimes offer different balances between features,
+performance, and security. 
+
+Kubernetes allows integration various container runtimes via a container
+runtime interface (CRI). |CL| supports Kubernetes with the various runtimes
+below:
+
+* `Docker (Dockershim)`_ with or without `Kata Containers`_.
```

Comment:
This is kind of a weird way to state with or without Kata containers for Docker.  Maybe for each of these say. It is just a suggestion.

Docker with fixed OCI runtime by Node
containerd with selectable OCI runtime by pod (runc or Kata Containers)
CRI-O with  selectable OCI runtime by pod (runc or Kata containers)

Reply:
I tried wording this a lot of different ways and am open to changing it. 

From a new reader perspective, it has too many technical details which don't really help in context of introducing the runtimes.  The details you mentioned are further in the sections after the different options are introduced.